lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -334,6 +334,17 @@ def gem_version_compare(first, second)
             Gem::Version.new(first.to_s) <=> Gem::Version.new(second.to_s)
           end
 
+          apipie :method, 'Returns content of Foreman SSL certificate' do
+            example 'foreman_ssl_certificate # => Content of file'
+          end
+          def foreman_ssl_certificate
```

Comment:
As pointed out elsewhere: the name of the macro doesn't match the content of the macro. Foreman doesn't have any setting now to figure out its own CA since that's only served by Apache or whatever webserver you're using.

Apache can be configured to send `SSL_SERVER_CERT` (the PEM encoded server certificate). It does have the downside that a certificate can be rotated while the CA typically isn't. So there are more failure scenarios to using it.

https://httpd.apache.org/docs/current/mod/mod_ssl.html

Reply:
There's another downside. I couldn't workaround so far, on debian system, curl --cacert does not accept such certificate. It may be my hands too but it seems it needs full path (multiple certs). From your other comments, I'm not sure if you're OK with using the suggested setting and ask for changing the description or you request changing the whole thing and get the actual cert for HTTP headers. Could you please clarify?

Reply:
> curl --cacert does not accept such certificate

With such, do you mean the server certificate, not the CA?

> From your other comments, I'm not sure if you're OK with using the suggested setting and ask for changing the description or you request changing the whole thing and get the actual cert for HTTP headers. Could you please clarify?

I think using the current setting is a good start and will work for a lot of cases. but `foreman_ssl_certificate` is an imprecise name that can only create confusion in its current implementation. I'm struggling to come up with a better name/description.

Reply:
> With such, do you mean the server certificate, not the CA?

Yes

> I think using the current setting is a good start and will work for a lot of cases. but foreman_ssl_certificate is an imprecise name that can only create confusion in its current implementation. I'm struggling to come up with a better name/description.

Ok. If we know, this should in the longer term be the foreman cert or foreman ca cert, I'd keep the name or call it `foreman_server_ca_cert`. User may want to customize this, for now they can edit the template and put it there. I guess we won't improve it until we refactor the whole cert handling.

Reply:
Perhaps open an issue and add a TODO in the code with that issue? One mitigating factor is that custom certs are often used to get a trusted certificate (like Let's Encrypt) or some corporate CA so it's probably not too bad. There's a decent chance those certificates are already in place on the target machine.

Reply:
I think we simply need the ca cert here. I think it‘s reasonable to introduce a setting for this, we can iterate on that later. The macro should clearly state that it returns the server ca cert.
The current approach is just technical debt that will do more harm than good. Let‘s not get carried away here.